tests/entrypoints/openai/test_completion_with_function_calling.py

```diff
@@ -35,9 +39,53 @@ async def client(server):
         yield async_client
 
 
+class TestCase(NamedTuple):
+    model_name: str
+    stream: bool
+    tool_choice: str
+    enable_thinking: bool
+
+
 @pytest.mark.asyncio
-@pytest.mark.parametrize("model_name", [MODEL_NAME])
-async def test_required_tool_use(client: openai.AsyncOpenAI, model_name: str):
+@pytest.mark.parametrize(
+    "test_case",
+    [
+        TestCase(model_name=MODEL_NAME,
```

Comment:
shall we just add more parameterize on "stream", "tool_choice" and "enable_thinking" instead of enumerate all of them here? :-)